File: src/main/java/org/example/kalkulationsprogramm/repository/SachkontoRepository.java
package org.example.kalkulationsprogramm.repository;

import org.example.kalkulationsprogramm.domain.Sachkonto;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface SachkontoRepository extends JpaRepository<Sachkonto, Long> {
    List<Sachkonto> findAllByOrderBySortierungAscBezeichnungAsc();
    List<Sachkonto> findByAktivTrueOrderBySortierungAscBezeichnungAsc();
}

File: src/main/kotlin/org/example/kalkulationsprogramm/repository/SachkontoRepository.kt
package org.example.kalkulationsprogramm.repository

import java.util.Optional
import org.example.kalkulationsprogramm.domain.Sachkonto
import org.springframework.data.jpa.repository.JpaRepository

interface SachkontoRepository : JpaRepository<Sachkonto, Long> {
    fun findAllByOrderBySortierungAscBezeichnungAsc(): List<Sachkonto>

    fun findByAktivTrueOrderBySortierungAscBezeichnungAsc(): List<Sachkonto>

    fun findByNummer(nummer: String): Optional<Sachkonto>
}
